todo_app/app.py

```diff
@@ -1,13 +1,44 @@
 from flask import Flask
 from flask import render_template, request, redirect
 from todo_app.data.todo import ToDoCard, ViewModel, get_todo_cards, move_todo_card, create_todo_card
-import datetime,pytest
+import datetime, os, requests, json
+from flask_login import LoginManager, login_required, UserMixin, login_user, current_user
+from oauthlib.oauth2 import WebApplicationClient
+
+login_manager = LoginManager()
+
+@login_manager.unauthorized_handler
+def unauthenticated():
+    client = WebApplicationClient(os.environ.get('GITHUB_CLIENTID'))
+    auth_redirect = client.prepare_request_uri('https://github.com/login/oauth/authorize')
+
+    return redirect(auth_redirect)
+
+@login_manager.user_loader
+def load_user(user_id):
+    return User(user_id)
+
+
+class User(UserMixin):
+    def __init__(self, user_id):
+        self.id = user_id
+    @property
+    def role(self):
+        if self.id == 'lachlanpearse-mclaren':
+            return 'writer'
+        else:
+            return 'reader'
 
 def create_app():
+    
 
     app = Flask(__name__)
+    app.secret_key = os.urandom(24)
```

Comment:
This isn't ideal. The secret_key is used for encrypting/decrypting cookies. If it's randomly assigned on startup, then you can't have multiple servers handling the same cookie, plus you'll force a login every time the server restarts.

It should be an env var.